wordpress-mobile/WordPress-Android
# Replaces the different checks for jetpack branding visibility with a single flag

This PR unifies the logic of jetpack banner and badge visibility. This also adds `isAccessedViaWPComRest` to the current logic. 

Note: The banner was removed on People and Themes in https://github.com/wordpress-mobile/WordPress-Android/pull/16941 and https://github.com/wordpress-mobile/WordPress-Android/pull/16942, but they're still updated here. A simple conflict may occur between these PRs.

To test:
1. Launch the WordPress app.
2. Check these screens. The banner and badges should be visible,
- Activity Log, and Activity Log detail,
- App Settings,
- Me,
- HOME tab,
- Notifications,
- Notification Settings,
- People,
- Sharing,
- Reader, Reader detail, Reader search
- Stats,
- Themes
3. Launch the Jetpack app, and check screens in 2. The banner and badges should be not visible.

Test 2:
1. Launch the WordPress app.
2. Select a self-hosted site that's not Jetpack connected.
3. Ensure the banner or badge is not visible on any screen.

## Regression Notes
1. Potential unintended areas of impact
Visibility of jetpack-powered banners and badges.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually on emulator.

3. What automated tests I added (or what prevented me from doing so)
I updated unit tests in `MySiteViewModelTest`.

PR submission checklist:
